corehq/apps/repeaters/views.py

```diff
@@ -33,18 +36,30 @@ def make_repeater(self):
         return repeater
 
 
-class RepeatRecordView(View):
+class RepeatRecordView(LoginAndDomainMixin, View):
 
     urlname = 'repeat_record'
     http_method_names = ['get', 'post']
 
+    @staticmethod
+    def get_record_or_404(request, domain):
+        try:
+            record = RepeatRecord.get(request.GET.get('record_id'))
+        except ResourceNotFound:
+            raise Http404()
+
+        if record.domain != domain:
+            raise Http404()
+
+        return record
+
     def get(self, request, domain):
-        record = RepeatRecord.get(request.GET.get('record_id'))
+        record = self.get_record_or_404(request, domain)
         content_type = record.repeater.get_payload_generator(
             record.repeater.format_or_default_format()
         ).content_type
         try:
-            payload = record.get_payload()
+            payload = record.get_payload(record)
```

Comment:
Ah, no! Totally unintentional change, thanks for catching in review! (I was playing with some different changes on another branch, which I ended up not going with, and I just didn't properly/fully revert it.)